### PR TITLE
[Refactor] Remove set_seed

### PR DESCRIPTION
**Pull request by patrickvonplaten**

### CODE REVIEW

File: src/diffusers/pipelines/stochatic_karras_ve/pipeline_stochastic_karras_ve.py [hunk 1]
```diff
@@ -18,9 +16,6 @@ class KarrasVePipeline(DiffusionPipeline):
     differential equations." https://arxiv.org/abs/2011.13456
     """
 
-    unet: UNet2DModel
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/stochatic_karras_ve/pipeline_stochastic_karras_ve.py:21]: @anton-l think we don't want those to be class attributes no? Haven't seen this anywhere else I think

File: src/diffusers/schedulers/scheduling_sde_ve.py [hunk 2]
```diff
@@ -98,6 +100,9 @@ def get_adjacent_sigma(self, timesteps, t):
         raise ValueError(f"`self.tensor_format`: {self.tensor_format} is not valid.")
 
     def set_seed(self, seed):
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_sde_ve.py:102]: We should not work with `set_seed(...)` here as it goes against the `generator` design. IMO passing generators around is the correct thing to do here because:
- We cannot really pass seeds around and globally set `manual_seed(...)` every time: 
   - It's not a great design to retrieve the current seed from a generator and then pass this -> better to pass generator directly
   - Flax passes PNRG keys around that can be split -> we cannot split PyTorch seeds in the same way and we cannot pass a `seed` into a `torch.randn(...)` function -> so let's go for the generator here

File: src/diffusers/pipelines/stochatic_karras_ve/pipeline_stochastic_karras_ve.py (hunk 1 above)
Comment (anton-l) [reply src/diffusers/pipelines/stochatic_karras_ve/pipeline_stochastic_karras_ve.py:21]: Actually I wanted to add them everywhere else instead! They help with type hinting in PyCharm and VSCode, otherwise they get highlighted as missing, since we set class attributes implicitly in `self.register_modules()`

Comment (patrickvonplaten) [reply src/diffusers/pipelines/stochatic_karras_ve/pipeline_stochastic_karras_ve.py:21]: Ah yeah I see - good point. Had to look this up actually: https://stackoverflow.com/questions/53897054/python-class-members-type-hinting

Good for me then! Reverting this change

File: src/diffusers/schedulers/scheduling_sde_ve.py [hunk 3]
```diff
@@ -98,6 +100,9 @@ def get_adjacent_sigma(self, timesteps, t):
...
+        warnings.warn(
+            "The method `set_seed` is outdated. Please consider passing a generator instead.", DeprecationWarning
```
Comment (pcuenca) [inline src/diffusers/schedulers/scheduling_sde_ve.py:104]: Maybe suggest it will be removed in the future (we can use whatever version you think would be reasonable)

```suggestion
            "The method `set_seed` is deprecated and will be removed in version `0.4.0`. Please consider passing a generator instead.", DeprecationWarning
```

**Review by pcuenca — APPROVED** [review]
Looks good!

File: src/diffusers/pipelines/score_sde_ve/pipeline_score_sde_ve.py [hunk 4]
```diff
@@ -42,11 +42,11 @@ def __call__(self, batch_size=1, num_inference_steps=2000, generator=None, outpu
             # correction step
             for _ in range(self.scheduler.correct_steps):
                 model_output = self.unet(sample, sigma_t)["sample"]
-                sample = self.scheduler.step_correct(model_output, sample)["prev_sample"]
+                sample = self.scheduler.step_correct(model_output, sample, generator=generator)["prev_sample"]
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/score_sde_ve/pipeline_score_sde_ve.py:45]: let's pass the generator here

File: src/diffusers/pipelines/ddpm/pipeline_ddpm.py [hunk 5]
```diff
@@ -56,7 +56,7 @@ def __call__(self, batch_size=1, generator=None, output_type="pil", **kwargs):
             model_output = self.unet(image, t)["sample"]
 
             # 2. compute previous image: x_t -> t_t-1
-            image = self.scheduler.step(model_output, t, image)["prev_sample"]
+            image = self.scheduler.step(model_output, t, image, generator=generator)["prev_sample"]
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/ddpm/pipeline_ddpm.py:59]: DDPM scheduler is also stochastic

File: src/diffusers/schedulers/scheduling_sde_ve.py [hunk 6]
```diff
@@ -14,8 +14,10 @@
 
 # DISCLAIMER: This file is strongly influenced by https://github.com/yang-song/score_sde_pytorch
 
+import warnings
+
 # TODO(Patrick, Anton, Suraj) - make scheduler framework indepedent and clean-up a bit
```
Comment (anton-l) [inline src/diffusers/schedulers/scheduling_sde_ve.py:19]:
```suggestion
```
No longer relevant probably? :)

**Review by anton-l — APPROVED** [review]
Thanks for the updates! Let's see how the tests behave now
